### PR TITLE
Update starknet-devnet 

### PR DESCRIPTION
**Pull request by ClementWalter**
Time spent on this PR: 0.2

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Using starknet-devnet python package

## What is the new behavior?

Use newly released starknet-devnet in rust, using the official docker image instead.

Note: `deploy_externally_owned_account` is currently failing, see https://github.com/0xSpaceShard/starknet-devnet-rs/issues/153

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -78,7 +78,7 @@ build-sol:
 
 run:
 	mkdir -p deployments/starknet-devnet
-	poetry run starknet-devnet --lite-mode --seed 0 --dump-on exit --dump-path deployments/starknet-devnet/devnet.pkl --disable-rpc-request-validation --timeout 600
+	docker run -p 5050:5050 shardlabs/starknet-devnet-rs --seed 0
```
Comment (Eikix) [inline Makefile:81]: we are not dumping on exit anymore?

File: Makefile [hunk 2]
```diff
@@ -78,7 +78,7 @@ build-sol:
 
 run:
 	mkdir -p deployments/starknet-devnet
-	poetry run starknet-devnet --lite-mode --seed 0 --dump-on exit --dump-path deployments/starknet-devnet/devnet.pkl --disable-rpc-request-validation --timeout 600
```
Comment (Eikix) [inline Makefile:81]: is the disable-rpc-request-validation the cause of the cryptic rpc bugs?

**Review by Eikix — COMMENTED** [review]
just a question, lgtm

File: Makefile (hunk 2 above)
Comment (ClementWalter) [reply Makefile:81]: this option doesn't exist anymore

Comment (ClementWalter) [reply Makefile:81]: but it was for skipping the strong type checking and there was mistmatches between api spec, devnet version and starknet py back in the days

**Review by Eikix — APPROVED** [review]
lgtm

**Pull request merged**